Customize highlight templates to include opt out functionality.

taranjeet opened the pull request:
This PR should merged after https://github.com/open-craft/edx-theme/pull/55 

Reviewer: @pomegranited

taranjeet committed:
Add upstream templates for weekly highlight emails.

taranjeet committed:
Customize highlight templates to include opt out functionality.

pomegranited commented on openedx/core/djangoapps/schedules/templates/schedules/edx_ace/courseupdate/email/body.html:
```diff
@@ -30,6 +30,13 @@
 
+            <p>
+                {% blocktrans trimmed %}
+                    To opt out of these emails, visit <a href="{% url 'bulk_email_opt_out' token=username course_id=course_ids.0 %}" >here</a>
```
nit: could you please reword to be:
```
<a href="{% unsubscribe_url %}">Unsubscribe from these emails</a>.
```

pomegranited commented on openedx/core/djangoapps/schedules/templates/schedules/edx_ace/courseupdate/email/body.txt:
```diff
@@ -14,5 +14,9 @@ We want to let you know what you can look forward to in week {{ week_num }}:
 {% endblocktrans %}
+
+{% blocktrans trimmed %}
+To opt out of these emails, visit <a href="{% url 'bulk_email_opt_out' token=username course_id=course_ids.0 %}" >here</a>
```
Similar comment here (also, this is text, so can't have `<a href`):
```
Unsubscribe from these emails: {% unsubscribe_url %}
```

pomegranited reviewed (changes requested):
@taranjeet  I'm having trouble getting this change to work end-to-end on the hawthorn test sandbox server?  The emails I'm getting still use the default template, and don't show an opt-out link.  Have you managed to test it at all?

*Update* I moved the change to `lms/templates/ace_common/edx_ace/common/base_body.html`, to correspond to a similar file in edx-platform under [themes/red-theme/lms](https://github.com/edx/edx-platform/tree/master/themes/red-theme/lms/templates/ace_common/edx_ace/common/base_body.html), and I'm now able to see the change.

However, I think there's a bug in the link syntax which would be addressed by using the `unsubscribe_url` change I suggested, because if I use your change directly, I don't get an email at all.  But when I do this, I do:
```
{% blocktrans trimmed %}
Unsubscribe from these emails {{ username }}
{% endblocktrans %}
```

taranjeet committed:
Change email unsubscribe text.

pomegranited commented:
@taranjeet  As per my comment above, can you please move this change into the footer for `lms/templates/ace_common/edx_ace/common/base_body.html`?  The `openedx` dir doesn't seem to be picked up by the comprehensive theme template finder.

I've already [added this new file into the `cloudera-upstream-templates` branch](https://github.com/open-craft/edx-theme/commit/0f01491271dc1c8ff0e78ab69c389c1684a3c64e) as per the [instructions for adding new custom templates to this branch](https://github.com/open-craft/edx-theme/tree/cloudera#custom-templates), so if you rebase this PR, then all you should need to do is customise that template with this new unsubscribe link.

pomegranited committed:
Add upstream version of openedx/core/djangoapps/ace_common/templates/ace_common/edx_ace/common/base_body.html

pomegranited committed:
Merge branch 'cloudera-upstream-templates' into cloudera

taranjeet committed:
Remove template change of openedx directory.

taranjeet committed:
Add unsubscribe url if present in present.

taranjeet commented:
@pomegranited : I have addressed the review.

pomegranited commented on lms/templates/ace_common/edx_ace/common/base_body.html:
```diff
@@ -0,0 +1,202 @@
+                        <tr>
+                            <td>
+                                {% blocktrans trimmed %}
+                                    Unsubscribe from these emails {{unsubscribe_url}}
```
Could you make this be a link again?  
```
<a href="{{unsubscribe_url}}">Unsubscribe from these emails</a>.
```

pomegranited reviewed (changes requested):
Thanks @taranjeet -- I'll deploy this to the test server and give it a try.  One comment below, and could you rebase this on the latest `cloudera` branch so we can see the diffs between this and the base template?

Thanks!

taranjeet committed:
Make unsubscribe url a link again.

taranjeet commented:
@pomegranited : I have rebased it before only. Just that the base branch need to be updated for the pull request. I have done it now.
Also I have addressed review concern as well. Please review.

pomegranited commented:
Template updated in https://github.com/edx/edx-platform/pull/19018, no need to customise it here.
